src/Controller/PageAdminController.php

```diff
@@ -38,6 +38,18 @@
  */
 final class PageAdminController extends CRUDController
 {
+    private BlockServiceManagerInterface $blockServiceManager;
+    private TemplateManagerInterface $templateManager;
+
+    public function __construct(
+        BlockServiceManagerInterface $blockServiceManager,
+        TemplateManagerInterface $templateManager
+    ) {
+        $this->blockServiceManager = $blockServiceManager;
+        $this->templateManager = $templateManager;
+
+    }
+
```

Comment:
IMO, we should not change how we load things inside CRUDController, we use the container, with the function below `getSubscribedServices`

Reply:
hmmm then we do not inject anything into the controllers' construct 👀

Reply:
We don't inject on the CRUDController because people are extending them, and we let them the constructor free.

Here the class is final so we could inject, but I agree that since it's already has a SubscribedServices we can keep using it for consistency.